src/test/java/org/candlepin/model/PoolTest.java

```diff
@@ -295,4 +303,128 @@ public void testIsNotDerivedPool() {
 
         assertFalse(pool.isDerived());
     }
+
+    @Nested
+    @DisplayName("Field Access Tests")
+    public class FieldAccessTest extends DatabaseTestFixture {
+
+        @Test
+        public void testGetProductUuidWithDesyncedProduct() throws Exception {
+            Owner owner = this.createOwner();
+            Product product = this.createProduct(TestUtil.randomString("id-"),
+                TestUtil.randomString("name-"),
+                owner);
+            Pool pool = this.createPool(owner, product);
+
+            // We don't want to persist the upcoming changes to the pool, so commit now
+            this.getCurrentSession().getTransaction().commit();
+            this.getCurrentSession().clear();
```

Comment:
You can use `this.getEntityManager().clear()` instead, or the `.clear()` method in any curator -- many of which are provided by the test fixture. For instance, `poolCurator.clear()`

Reply:
Updated